float32 setPoint
float64 time